app/uk/gov/hmrc/merchandiseinbaggage/controllers/Navigator.scala

```diff
@@ -107,4 +122,23 @@ object Navigator {
       case Amend => RetrieveDeclarationController.onPageLoad()
     }
 
+  private def reviewGoodsController(
+    declareMoreGoods: YesNo,
+    declarationJourney: DeclarationJourney,
+    upsert: DeclarationJourney => Future[DeclarationJourney])(implicit ec: ExecutionContext): Future[Call] =
+    declareMoreGoods match {
+      case Yes => updateEntriesAndRedirect(declarationJourney, upsert)
+      case No  => Future.successful(PaymentCalculationController.onPageLoad())
+    }
+
+  private def updateEntriesAndRedirect(declarationJourney: DeclarationJourney, upsert: DeclarationJourney => Future[DeclarationJourney])(
```

Comment:
i would move these operations outside of this class as it should not deal these type of operations

Reply:
done

Reply:
approved, but i actually meant move `def updateEntriesAndRedirect` completely out of this Navigator class, because this class just deals with navitations